packages/map-toolkit/src/deckgl/hooks/use-map-events.ts

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright 2025 Hypergiant Galactic Systems Inc. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+import { useOn } from '@accelint/bus/react';
+import { MapEvents } from '../base-map/events';
+import type {
+  MapClickEvent,
+  MapClickPayload,
+  MapHoverEvent,
+  MapHoverPayload,
+} from '../base-map/types';
+
+type UseMapEventsProps = {
+  onHover?: (payload: MapHoverPayload) => void;
+  onClick?: (payload: MapClickPayload) => void;
+};
+
+export function useMapEvents({ onHover, onClick }: UseMapEventsProps) {
```

Comment:
I think I want to get rid of this hook. Because it always sets up listeners, even if you're not using them. And if we expand on that, we'll just keep adding more listeners and that will multiply with each usage. I think we need to enforce the usage of `useOn` directly